### DISCUSSION TITLE
Nest thermostat fan status

### DISCUSSION THREAD

**Discussion by dadsalleb** [potmat/homebridge-google-nest-sdm#119] [Q&A]
Hello, I installed GoogleNest and tested a bit.
It looks like I can see the status of room temperature and target temperature from Home app. The status changes properly when I change it via wall mounted thermostat device. However, it doesn't look like the fan running status does appear correctly. I used 15 min per an hour of fan running schedule, so it furnance run the fan automatically. The status can be checed in Google and Nest apps but I can't check it on Home app. I can control the fan via Home App but I can't check the status of fan properly.

**Comment by dthorndyke** [comment]
Hi, I’m noticing the same issue. The fan switch is not showing the correct “on” status during the times when the fan is running as scheduled via the “Fan Schedule” feature in the Nest app. 

When manually turning the thermostat fan on (through the Nest app), the fan status in homekit doesn’t update for a minute or two. When manually turning the fan off, it updates pretty immediately.

Using an automation triggered when the thermostat fan runs, I was trying to sync a booster fan to the times why my furnace fan is running. We have the Fan Schedule enabled to help circulate air throughout the house since the temperatures can get quite uneven in certain rooms.

Does anyone know if the incorrect status during scheduled fan events is a limitation of the API or perhaps some other cause?

Thanks in advance!

**Reply by dthorndyke** [reply]
Ah, I am looking at the [SDM docs](https://developers.google.com/nest/device-access/traits/device/fan) now and it appears that they only expose these details about the thermostat fan:
- `timerMode` - Values: "ON", “OFF"
- `timerTimeout` - Timestamp, in RFC 3339 format, at which timer mode will turn to OFF.

If these docs contain the entire list of details that can be fetched about the thermostat, then it looks like Google simply doesn’t expose the fan status when it is running due to the Fan Schedule feature. That’s a bit baffling to me...
